### PR TITLE
[AppKit] Xcode 14 Beta 1-5

### PR DESCRIPTION
**Pull request by chamons**

### CODE REVIEW

File: src/appkit.cs [hunk 1]
```diff
@@ -6973,6 +6991,12 @@ partial interface NSFont : NSSecureCoding, NSCopying {
 		[Mac (11,0)]
 		[Export ("fontWithSize:")]
 		NSFont GetFont (nfloat fontSize);
+
+		[Advice ("Use constants from 'NSFontWeight' and 'NSFontWidth'.")]
+		[Mac (13,0)]
+		[Static]
+		[Export ("systemFontOfSize:weight:width:")]
+		NSFont SystemFontOfSize (nfloat fontSize, double weight, double width);
```
Comment (dalexsoto) [inline src/appkit.cs:6999]:
```suggestion
		NSFont GetSystemFontOfSize (nfloat fontSize, double weight, double width);
```

File: src/appkit.cs [hunk 2]
```diff
@@ -7427,6 +7451,25 @@ interface NSFontWeight {
...
 	}
 
+	[Mac (10, 10)]
+	[NoMacCatalyst]
+	[Static]
+	interface NSFontWidth
+	{
+		[Mac (13, 0)]
+		[Field ("NSFontWidthCompressed")]
+		nfloat Compressed { get; }
```
Comment (dalexsoto) [inline src/appkit.cs:7461]: the only nfloat? looks funny...

File: src/appkit.cs [hunk 3]
```diff
@@ -9912,6 +9955,18 @@ partial interface NSImage : NSCopying, NSSecureCoding, NSPasteboardReading, NSPa
 		[NoMacCatalyst]
 		[Export ("symbolConfiguration", ArgumentSemantic.Copy)]
 		NSImageSymbolConfiguration SymbolConfiguration { get; }
+
+		[NoMacCatalyst, Mac (13,0)]
+		[Static]
+		[Export ("imageWithSymbolName:variableValue:")]
+		[return: NullAllowed]
+		NSImage ImageWithSymbolName (string name, double value);
```
Comment (dalexsoto) [inline src/appkit.cs:9963]:
```suggestion
		NSImage GetImage (string symbolName, double variableValue);
```

File: src/appkit.cs [hunk 4]
```diff
@@ -9912,6 +9955,18 @@ partial interface NSImage : NSCopying, NSSecureCoding, NSPasteboardReading, NSPa
...
+
+		[NoMacCatalyst, Mac (13,0)]
+		[Static]
+		[Export ("imageWithSystemSymbolName:variableValue:accessibilityDescription:")]
+		[return: NullAllowed]
+		NSImage ImageWithSystemSymbolName (string name, double value, [NullAllowed] string description);
```
Comment (dalexsoto) [inline src/appkit.cs:9969]:
```suggestion
		NSImage GetImage (string systemSymbolName, double variableValue, [NullAllowed] string accessibilityDescription);
```

File: src/appkit.cs [hunk 5]
```diff
@@ -20154,6 +20112,15 @@ interface NSToolbarDelegate {
 
 		[Export ("toolbarDidRemoveItem:"), EventArgs ("NSNotification")]
 		void DidRemoveItem (NSNotification notification);
+
+		[Mac (13,0), MacCatalyst (16,0), DelegateName ("NSToolbarImmovableItemIdentifiers"), DefaultValue (null)]
+		[Export ("toolbarImmovableItemIdentifiers:")]
+		NSSet<NSString> ToolbarImmovableItemIdentifiers (NSToolbar toolbar);
```
Comment (dalexsoto) [inline src/appkit.cs:20118]:
```suggestion
		NSSet<NSString> GetToolbarImmovableItemIdentifiers (NSToolbar toolbar);
```

File: src/appkit.cs [hunk 6]
```diff
@@ -20154,6 +20112,15 @@ interface NSToolbarDelegate {
...
+
+		[Mac (13,0), MacCatalyst (16,0)]
+		[Export ("toolbar:itemIdentifier:canBeInsertedAtIndex:"), DelegateName ("NSToolbarCanInsert"), DefaultValue (true)]
+		bool ItemCanBeInsertedAt (NSToolbar toolbar, string itemIdentifier, nint index);
```
Comment (dalexsoto) [inline src/appkit.cs:20122]:
```suggestion
		bool GetItemCanBeInsertedAt (NSToolbar toolbar, string itemIdentifier, nint index);
```

File: src/appkit.cs [hunk 7]
```diff
@@ -28450,5 +28426,88 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
 		[Mac (12,0)]
 		[Export ("configurationByApplyingConfiguration:")]
 		NSImageSymbolConfiguration Create (NSImageSymbolConfiguration configuration);
+
+		[Mac (13,0)]
+		[Static]
+		[Export ("configurationPreferringMonochrome")]
+		NSImageSymbolConfiguration ConfigurationPreferringMonochrome ();
```
Comment (dalexsoto) [inline src/appkit.cs:28433]:
```suggestion
		NSImageSymbolConfiguration GetConfigurationPreferringMonochrome ();
```

File: src/appkit.cs [hunk 8]
```diff
@@ -28450,5 +28426,88 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
...
+
+		[Mac (13,0)]
+		[Static]
+		[Export ("configurationPreferringHierarchical")]
+		NSImageSymbolConfiguration ConfigurationPreferringHierarchical ();
```
Comment (dalexsoto) [inline src/appkit.cs:28438]:
```suggestion
		NSImageSymbolConfiguration GetConfigurationPreferringHierarchical ();
```

File: src/appkit.cs [hunk 9]
```diff
@@ -28450,5 +28426,88 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
...
+	[BaseType (typeof(NSControl))]
+	interface NSComboButton
+	{
+		[DesignatedInitializer]
+		[Export ("initWithFrame:")]
+		NativeHandle Constructor (CGRect frameRect);
+
+		[Static]
+		[Export ("comboButtonWithTitle:menu:target:action:")]
+		NSComboButton CreateWithTitle (string title, [NullAllowed] NSMenu menu, [NullAllowed] NSObject target, [NullAllowed] Selector action);
```
Comment (dalexsoto) [inline src/appkit.cs:28451]:
```suggestion
		NSComboButton Create (string title, [NullAllowed] NSMenu menu, [NullAllowed] NSObject target, [NullAllowed] Selector action);
```

File: src/appkit.cs [hunk 10]
```diff
@@ -28450,5 +28426,88 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
...
+
+		[Static]
+		[Export ("comboButtonWithImage:menu:target:action:")]
+		NSComboButton CreateWithImage (NSImage image, [NullAllowed] NSMenu menu, [NullAllowed] NSObject target, [NullAllowed] Selector action);
```
Comment (dalexsoto) [inline src/appkit.cs:28455]:
```suggestion
		NSComboButton Create (NSImage image, [NullAllowed] NSMenu menu, [NullAllowed] NSObject target, [NullAllowed] Selector action);
```

File: src/appkit.cs [hunk 11]
```diff
@@ -28450,5 +28426,88 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
...
+
+		[Static]
+		[Export ("comboButtonWithTitle:image:menu:target:action:")]
+		NSComboButton CreateWithTitle (string title, NSImage image, [NullAllowed] NSMenu menu, [NullAllowed] NSObject target, [NullAllowed] Selector action);
```
Comment (dalexsoto) [inline src/appkit.cs:28459]:
```suggestion
		NSComboButton Create (string title, NSImage image, [NullAllowed] NSMenu menu, [NullAllowed] NSObject target, [NullAllowed] Selector action);
```

File: src/appkit.cs [hunk 12]
```diff
@@ -28450,5 +28426,88 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
...
+
+		[Export ("style", ArgumentSemantic.Assign)]
+		NSComboButtonStyle Style { get; set; }
 	}
+
+	interface INSPreviewRepresentableActivityItem { }
+
+	[NoMacCatalyst, Mac (13,0)]
+	[Protocol]
+	[BaseType (typeof(NSObject))]
```
Comment (dalexsoto) [inline src/appkit.cs:28481]: This needs to be removed or the ModelAttribute, I think the former is the right choice 
```suggestion
```

File: src/xkit.cs [hunk 13]
```diff
@@ -3963,4 +3974,153 @@ interface NSTextContentStorage : NSTextStorageObserving
...
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] NSTextContentManager textContentManager);
+
+		[Export ("initWithParentElement:textList:contents:markerAttributes:childElements:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] NSTextListElement parent, NSTextList textList, [NullAllowed] NSAttributedString contents, [NullAllowed] NSDictionary<NSString, NSObject> markerAttributes, [NullAllowed] NSTextListElement[] children);
+
+		[Static]
+		[Export ("textListElementWithContents:markerAttributes:textList:childElements:")]
+		NSTextListElement CreateWithContents (NSAttributedString contents, [NullAllowed] NSDictionary<NSString, NSObject> markerAttributes, NSTextList textList, [NullAllowed] NSTextListElement[] children);
```
Comment (dalexsoto) [inline src/xkit.cs:4039]:
```suggestion
		NSTextListElement Create (NSAttributedString contents, [NullAllowed] NSDictionary markerAttributes, NSTextList textList, [NullAllowed] NSTextListElement[] children);
```

File: src/xkit.cs [hunk 14]
```diff
@@ -3963,4 +3974,153 @@ interface NSTextContentStorage : NSTextStorageObserving
...
+
+		[Static]
+		[Export ("textListElementWithChildElements:textList:nestingLevel:")]
+		[return: NullAllowed]
+		NSTextListElement CreateWithChildElements (NSTextListElement[] children, NSTextList textList, nint nestingLevel);
```
Comment (dalexsoto) [inline src/xkit.cs:4044]:
```suggestion
		NSTextListElement Create (NSTextListElement[] children, NSTextList textList, nint nestingLevel);
```

File: src/xkit.cs [hunk 15]
```diff
@@ -3963,4 +3974,153 @@ interface NSTextContentStorage : NSTextStorageObserving
...
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] NSAttributedString attributedString);
+
+		[Export ("initWithTextContentManager:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] NSTextContentManager textContentManager);
+
+		[Export ("initWithParentElement:textList:contents:markerAttributes:childElements:")]
+		[DesignatedInitializer]
+		NativeHandle Constructor ([NullAllowed] NSTextListElement parent, NSTextList textList, [NullAllowed] NSAttributedString contents, [NullAllowed] NSDictionary<NSString, NSObject> markerAttributes, [NullAllowed] NSTextListElement[] children);
```
Comment (dalexsoto) [inline src/xkit.cs:4035]: I would probably not use the generic dictionary here so we can create a better API in the future

```suggestion
		NativeHandle Constructor ([NullAllowed] NSTextListElement parent, NSTextList textList, [NullAllowed] NSAttributedString contents, [NullAllowed] NSDictionary markerAttributes, [NullAllowed] NSTextListElement[] children);
```

File: src/xkit.cs [hunk 16]
```diff
@@ -3963,4 +3974,153 @@ interface NSTextContentStorage : NSTextStorageObserving
...
+		NSTextListElement CreateWithChildElements (NSTextListElement[] children, NSTextList textList, nint nestingLevel);
+
+		[Export ("textList", ArgumentSemantic.Strong)]
+		NSTextList TextList { get; }
+
+		[NullAllowed, Export ("contents", ArgumentSemantic.Strong)]
+		NSAttributedString Contents { get; }
+
+		[NullAllowed, Export ("markerAttributes", ArgumentSemantic.Strong)]
+		NSDictionary<NSString, NSObject> MarkerAttributes { get; }
```
Comment (dalexsoto) [inline src/xkit.cs:4053]: Probably we can create a strong dictionary in the future

```suggestion
		NSDictionary WeakMarkerAttributes { get; }
```

**Review by dalexsoto — CHANGES_REQUESTED** [review]

File: src/appkit.cs (hunk 2 above)
Comment (chamons) [reply src/appkit.cs:7461]: That does seem wrong, looking.

File: src/xkit.cs [hunk 17]
```diff
@@ -3963,4 +3974,153 @@ interface NSTextContentStorage : NSTextStorageObserving
...
+
+		[Field ("NSTextListMarkerDecimal")]
+		Decimal,
+	}
+
+	[TV (16,0), NoWatch, Mac (10,10), iOS (16,0)]
+	[Flags]
+	[Native]
+	public enum NSTextListOptions : ulong {
+		PrependEnclosingMarker = 1
```
Comment (rolfbjarne) [inline src/xkit.cs:4124]:
```suggestion
		PrependEnclosingMarker = 1,
```

File: src/xkit.cs [hunk 18]
```diff
@@ -3963,4 +3974,153 @@ interface NSTextContentStorage : NSTextStorageObserving
...
+		LowercaseRoman,
+
+		[Field ("NSTextListMarkerUppercaseRoman")]
+		UppercaseRoman,
+
+		[Field ("NSTextListMarkerDecimal")]
+		Decimal,
+	}
+
+	[TV (16,0), NoWatch, Mac (10,10), iOS (16,0)]
```
Comment (rolfbjarne) [inline src/xkit.cs:4120]: MacCatalyst?

File: tests/xtro-sharpie/api-annotations-dotnet/macOS-AppKit.ignore [hunk 19]
```diff
@@ -1197,3 +1197,16 @@
...
+!deprecated-attribute-missing! NSColorWell::setBordered: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSImage::lockFocus missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSImage::lockFocusFlipped: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSImage::unlockFocus missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSToolbar::centeredItemIdentifier missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSToolbar::setCenteredItemIdentifier: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSToolbar::setSizeMode: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSToolbar::sizeMode missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSWindow::setShowsToolbarButton: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSWindow::showsToolbarButton missing a [Deprecated] attribute
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/api-annotations-dotnet/macOS-AppKit.ignore:1212]: 👀

File: tests/xtro-sharpie/macOS-AppKit.ignore [hunk 20]
```diff
@@ -1318,3 +1318,15 @@
...
+!deprecated-attribute-missing! NSColorWell::setBordered: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSImage::lockFocus missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSImage::lockFocusFlipped: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSImage::unlockFocus missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSToolbar::centeredItemIdentifier missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSToolbar::setCenteredItemIdentifier: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSToolbar::setSizeMode: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSToolbar::sizeMode missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSWindow::setShowsToolbarButton: missing a [Deprecated] attribute
+!deprecated-attribute-missing! NSWindow::showsToolbarButton missing a [Deprecated] attribute
```
Comment (rolfbjarne) [inline tests/xtro-sharpie/macOS-AppKit.ignore:1332]: 👀

File: src/appkit.cs [hunk 21]
```diff
@@ -6973,6 +6991,12 @@ partial interface NSFont : NSSecureCoding, NSCopying {
 		[Mac (11,0)]
 		[Export ("fontWithSize:")]
 		NSFont GetFont (nfloat fontSize);
+
+		[Advice ("Use constants from 'NSFontWeight' and 'NSFontWidth'.")]
+		[Mac (13,0)]
+		[Static]
+		[Export ("systemFontOfSize:weight:width:")]
+		NSFont GetSystemFontOfSize (nfloat fontSize, double weight, double width);
```
Comment (rolfbjarne) [inline src/appkit.cs:6999]: Fonts need some custom treatment (look at the UIKit PR for the same selector): 

https://github.com/xamarin/xamarin-macios/pull/15694/files#diff-97d65d3ebcf51cbd79386a15db035a4a338cc9bf0fdcb2cd653d6b015ecd1770R6226

File: src/appkit.cs [hunk 22]
```diff
@@ -28450,5 +28426,87 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
 		[Mac (12,0)]
 		[Export ("configurationByApplyingConfiguration:")]
 		NSImageSymbolConfiguration Create (NSImageSymbolConfiguration configuration);
+
+		[Mac (13,0)]
+		[Static]
+		[Export ("configurationPreferringMonochrome")]
+		NSImageSymbolConfiguration GetConfigurationPreferringMonochrome ();
```
Comment (rolfbjarne) [inline src/appkit.cs:28433]: All the other methods in this class seem to use Create:

```suggestion
		NSImageSymbolConfiguration CreateConfigurationPreferringMonochrome ();
```

File: src/appkit.cs [hunk 23]
```diff
@@ -28450,5 +28426,87 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
...
+
+		[Mac (13,0)]
+		[Static]
+		[Export ("configurationPreferringHierarchical")]
+		NSImageSymbolConfiguration GetConfigurationPreferringHierarchical ();
```
Comment (rolfbjarne) [inline src/appkit.cs:28438]: Same:

```suggestion
		NSImageSymbolConfiguration CreateConfigurationPreferringHierarchical ();
```

**Review by rolfbjarne — CHANGES_REQUESTED** [review]

File: src/appkit.cs [hunk 24]
```diff
@@ -20130,6 +20078,16 @@ partial interface NSToolbar {
...
+
+		[Mac (13, 0), MacCatalyst (16, 0)]
+		[Export ("centeredItemIdentifiers", ArgumentSemantic.Copy)]
+		NSSet<NSString> CenteredItemIdentifiers { get; set; }
+
+		[NoMacCatalyst][Mac (13, 0)]
+		[Field ("NSToolbarItemKey")]
+		NSString NSToolbarItemKey { get; }
+
+
```
Comment (tj-devel709) [inline src/appkit.cs:20090]:
```suggestion
```
nit: extra new lines

File: src/appkit.cs [hunk 25]
```diff
@@ -28450,5 +28426,87 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
...
+		NSImageSymbolConfiguration GetConfigurationPreferringMonochrome ();
+
+		[Mac (13,0)]
+		[Static]
+		[Export ("configurationPreferringHierarchical")]
+		NSImageSymbolConfiguration GetConfigurationPreferringHierarchical ();
+	}
+
+	[NoMacCatalyst, Mac (13,0)]
+	[BaseType (typeof(NSControl))]
```
Comment (tj-devel709) [inline src/appkit.cs:28442]:
```suggestion
	[BaseType (typeof (NSControl))]
```
nit

File: src/appkit.cs [hunk 26]
```diff
@@ -28450,5 +28426,87 @@ interface NSImageSymbolConfiguration : NSCopying, NSSecureCoding
...
+
+		[NullAllowed, Export ("imageProvider", ArgumentSemantic.Strong)]
+		NSItemProvider ImageProvider { get; }
+
+		[NullAllowed, Export ("iconProvider", ArgumentSemantic.Strong)]
+		NSItemProvider IconProvider { get; }
+	}
+
+	[NoMacCatalyst, Mac (13,0)]
+	[BaseType (typeof(NSObject))]
```
Comment (tj-devel709) [inline src/appkit.cs:28498]:
```suggestion
	[BaseType (typeof (NSObject))]
```

**Review by tj-devel709 — APPROVED** [review]
Aside from the small nit things I mentioned and others' suggestions, LGTM!

File: src/appkit.cs [hunk 27]
```diff
@@ -6935,6 +6953,13 @@ partial interface NSFont : NSSecureCoding, NSCopying {
 		[Internal]
 		IntPtr _SystemFontOfSize (nfloat fontSize, nfloat weight);
 
+		[Advice ("Use constants from 'NSFontWeight' and 'NSFontWidth'.")]
```
Comment (rolfbjarne) [inline src/appkit.cs:6956]: I believe [Advice] is useless on an [Internal] method.

```suggestion
		[Advice ("Use constants from 'NSFontWeight' and 'NSFontWidth'.")]
```

**Review by rolfbjarne — APPROVED** [review]

**Review by dalexsoto — APPROVED** [review]

**Pull request merged**